### ISSUE TITLE
Improve overall programming style

### ISSUE THREAD

**Issue by MatthewRock** [agrentum/GITSP#1]
Hello!
I'm writing this, because I'm worried some beginners might see this code and find it useful without reading other resources, which might be bad for them.

The code you are using is fine, but sometimes it isn't "lispy" enough(isn't using the powers of lisp, or is using old ways of doing things).
For example, code from chapter1:

(defun CUSTOM-ABS (number)
  (if (> number 0) number
      (- 0 number)))

could be changed to:

(defun CUSTOM-ABS (number)
  (if (plusp number) number
      (- number)))

It's a bit simpler to write, and same if not easier to read. And this doesn't teach you some bad things(like you don't need 0 if you want to have negative value of number).
There are some other things that aren't really good, like twop:

(defun TWOP (number)
  (equal number 2))

Now I know that comment stated that this works only for integers, but this code is just wrong. For starters, as a rule of thumb - you should compare numbers with (function =). This is meant to compare numbers, so let's stay with it.
But there's more - 2 and 2.0 should both pass test - because they are both 2. 2/1 is also equal to 2, but it will past the test as well - because equal tests for type of object, and they all are of a different type. This example teaches to use equal when you shouldn't use it. I think it's bad.
Also, in fun.lisp you're using setq. It's kind of old and shouldn't be used without prior explanation about how it works, because sometimes it won't work like you would want it to work. You should use(or teach beginners to use) setf instead. See http://stackoverflow.com/questions/971734/lisp-style-setq-vs-setf and http://stackoverflow.com/questions/869529/difference-between-set-setq-and-setf-in-common-lisp to know the difference.

So, to sum things up: I don't want to flame, or to criticize your work(too much :)), but I think that this is an issue. Some beginners might find it, and learn bad things from this. And we all know that it's harder to get rid of bad habits, then to learn them.
So I would suggest that either you fix the code, ask me for fixing the code(which I can do), or just remove it, if you don't plan on doing anything with it. It might be harmful, and I just care about beginners, because Common Lisp has some tricks that make it tricky.

**Comment by agrentum** [comment]
Hi, sorry for late response. I will correct the mistakes and redo the intro anyway. As you already know, I'm a beginner and simply wanted to provide some simple utilities that took me a long time to grasp. Thank you for your pointers, sources and would love to get some help from you.

Sorry for not responding earlier, I was a bit biased toward 'feedback' I got from few people earlier via email and reddit PMs. After getting fourth message where most coherent sentence was 'this is shit, go away from programming you moron!' I had high reservations about reading this issue. After finally reading it, I must say your issue report is most kind and I would be honoured to get help from you, if your offer still stands. Even if it would end up with you working on fork as some 'bad style' example, after I would expunge all materials on my part.

**Comment by priyadarshan** [comment]
@agrentum Very well spoken. I personally found your code very helpful. Thanks so much for sharing it.

**Comment by MatthewRock** [comment]
I hope I didn't sound rude. I just wanted to make sure this won't misguide anyone. :)
I can help, although I'm not any Lisp guru - I just read some books, wrote some simple programs for me to use - nothing big, I'm far from being Xach - but I can spot some weak points and help you improve them.
If you're willing to improve it, contact me, and I'll look into it again(now I forgot what was wrong there - it has been a while :)).

**Comment by agrentum** [comment]
@priyadarshan thank you, but while I am enthusiastic about the topic I don't think I am a good teacher. Past decade I was coding almost exclusively in C, Mathematica and Fortran and I treat programming as utility and it is all that I could describe as my programming experience. I'm trying to improve my understanding of Lisp but between poor health and university work it is my side activity. At best.

@MatthewRock I will contact you with some preliminary corrections at Monday or Tuesday if it is not a problem. Sorry for making you wait, especially since you are the one helping me, but I have to finish my seminar project. As far as your first post goes, I did not read it as rude or condescending. You made your case, provided solid reasons with references and offered help. I'm working under different aliases here at github and this is almost literally the most concise, reasonable and weighted response in regard to my work I have ever received. I have had closed and deleted countless projects because of vicious comments toward me, not my work. I was sincere while stating it would be an honour for me to work with someone such as yourself. No matter how petty it sounds, I think that some people in here (github and reddit) got bullied to a point where they can't talk without insulting others.

In addition, excuse weird syntax, punctuation or grammar errors on my part. I do try to be careful but English is my second language.

**Comment by MatthewRock** [comment]
@agrentum 
Same here about English. I'm quite impressed that I find someone in Internet who isn't insulting everyone else, but it's a good sign for sure. I'm willing to improve your repo.
I will check the files tomorrow and try to improve them.
Also please forgive me if I don't send files too fast, but I'm currently using Windows - which is the worst workstation one could imagine - and can't get proper internet connection on my Linux, so I need to use USB instead of internet :).
PS. I hope your health will improve soon :)
